src/VkDispatchTable.h

```diff
@@ -0,0 +1,4617 @@
+#ifndef PFN_vkCmdDebugMarkerBeginEXT_LOAD
+#define PFN_vkCmdDebugMarkerBeginEXT_LOAD
+		fp_vkCmdDebugMarkerBeginEXT = (PFN_vkCmdDebugMarkerBeginEXT)procAddr(device, "vkCmdDebugMarkerBeginEXT");
+#endif
+#ifndef PFN_vkCmdDebugMarkerEndEXT_LOAD
+#define PFN_vkCmdDebugMarkerEndEXT_LOAD
+		fp_vkCmdDebugMarkerEndEXT = (PFN_vkCmdDebugMarkerEndEXT)procAddr(device, "vkCmdDebugMarkerEndEXT");
+#endif
+#ifndef PFN_vkCmdDebugMarkerInsertEXT_LOAD
+#define PFN_vkCmdDebugMarkerInsertEXT_LOAD
+		fp_vkCmdDebugMarkerInsertEXT = (PFN_vkCmdDebugMarkerInsertEXT)procAddr(device, "vkCmdDebugMarkerInsertEXT");
+#endif
+#endif
+#if defined(VK_KHR_video_queue) && defined(VK_VERSION_1_0)
+#ifndef PFN_vkCreateVideoSessionKHR_LOAD
+#define PFN_vkCreateVideoSessionKHR_LOAD
+		fp_vkCreateVideoSessionKHR = (PFN_vkCreateVideoSessionKHR)procAddr(device, "vkCreateVideoSessionKHR");
+#endif
+#ifndef PFN_vkDestroyVideoSessionKHR_LOAD
+#define PFN_vkDestroyVideoSessionKHR_LOAD
+		fp_vkDestroyVideoSessionKHR = (PFN_vkDestroyVideoSessionKHR)procAddr(device, "vkDestroyVideoSessionKHR");
+#endif
+#ifndef PFN_vkGetVideoSessionMemoryRequirementsKHR_LOAD
+#define PFN_vkGetVideoSessionMemoryRequirementsKHR_LOAD
+		fp_vkGetVideoSessionMemoryRequirementsKHR = (PFN_vkGetVideoSessionMemoryRequirementsKHR)procAddr(device, "vkGetVideoSessionMemoryRequirementsKHR");
+#endif
+#ifndef PFN_vkBindVideoSessionMemoryKHR_LOAD
+#define PFN_vkBindVideoSessionMemoryKHR_LOAD
+		fp_vkBindVideoSessionMemoryKHR = (PFN_vkBindVideoSessionMemoryKHR)procAddr(device, "vkBindVideoSessionMemoryKHR");
+#endif
+#ifndef PFN_vkCreateVideoSessionParametersKHR_LOAD
+#define PFN_vkCreateVideoSessionParametersKHR_LOAD
+		fp_vkCreateVideoSessionParametersKHR = (PFN_vkCreateVideoSessionParametersKHR)procAddr(device, "vkCreateVideoSessionParametersKHR");
+#endif
+#ifndef PFN_vkUpdateVideoSessionParametersKHR_LOAD
+#define PFN_vkUpdateVideoSessionParametersKHR_LOAD
+		fp_vkUpdateVideoSessionParametersKHR = (PFN_vkUpdateVideoSessionParametersKHR)procAddr(device, "vkUpdateVideoSessionParametersKHR");
+#endif
+#ifndef PFN_vkDestroyVideoSessionParametersKHR_LOAD
+#define PFN_vkDestroyVideoSessionParametersKHR_LOAD
+		fp_vkDestroyVideoSessionParametersKHR = (PFN_vkDestroyVideoSessionParametersKHR)procAddr(device, "vkDestroyVideoSessionParametersKHR");
+#endif
+#ifndef PFN_vkCmdBeginVideoCodingKHR_LOAD
+#define PFN_vkCmdBeginVideoCodingKHR_LOAD
+		fp_vkCmdBeginVideoCodingKHR = (PFN_vkCmdBeginVideoCodingKHR)procAddr(device, "vkCmdBeginVideoCodingKHR");
+#endif
+#ifndef PFN_vkCmdEndVideoCodingKHR_LOAD
+#define PFN_vkCmdEndVideoCodingKHR_LOAD
+		fp_vkCmdEndVideoCodingKHR = (PFN_vkCmdEndVideoCodingKHR)procAddr(device, "vkCmdEndVideoCodingKHR");
+#endif
+#ifndef PFN_vkCmdControlVideoCodingKHR_LOAD
+#define PFN_vkCmdControlVideoCodingKHR_LOAD
+		fp_vkCmdControlVideoCodingKHR = (PFN_vkCmdControlVideoCodingKHR)procAddr(device, "vkCmdControlVideoCodingKHR");
+#endif
+#endif
+#if defined(VK_KHR_video_decode_queue) && defined(VK_VERSION_1_0)
```

Comment:
The version string is collected from the require nodes in extensions. I just set it to `VK_VERSION_1_0` when building the dictionary of extension commands so that there isn't any special branching needed in the final compilation.

Reply:
Then implement that logic in the autogen, it will make the resulting code much cleaner because then you don't have to rely on this 'guard in rare cases'. Autogen is powerful like that, you can work around potential problems in ways that with regular code is more obnoxious.